### DISCUSSION TITLE
Flat URL structure across my entire site, including the blog?

### DISCUSSION THREAD

**Discussion by futuresoup** [squidfunk/mkdocs-material#5298] [Q&A]
Is a completely flat URL structure possible? 

I'm able to do it for pages, but what about with the blog feature enabled?

**Is the `/blog/` part of the URL required**?

For example: `mysite.com/blog-post-title` instead of `mysite.com/blog/blog-post-title`?

My reasoning behind this is that I am perceiving a flat URL structure as the way to go for long term site planning to avoid any possibility of page redirections with subdirectory urls etc.

One potential downside is a messy root directory in the `docs` folder and a bunch of folders with index.html files in the `site` build. 

I'm not sure if that's necessarily a bad thing, I can live with it, if it means my URLs are sustainable and I can avoid messy complicated redirects years from now...

**Comment by squidfunk** [comment]
In theory this should be possibly by just naming your `.md` files exactly as the slugs you like to have. For the blog plugin, you can change [`blog_dir` and set it to `.`](https://squidfunk.github.io/mkdocs-material/setup/setting-up-a-blog/#standalone), which is effectively a standalone blog. Note that the `posts` directory must now be located directly in the `docs_dir`. [We also have a blog article that walks you through the process of creating a standalone blog](https://squidfunk.github.io/mkdocs-material/blog/2022/09/12/blog-support-just-landed/#creating-a-standalone-blog).
